src/transformers/trainer.py

```diff
@@ -292,13 +298,23 @@ def __init__(
             if isinstance(eval_dataset, datasets.Dataset):
                 self._remove_unused_columns(self.eval_dataset, description="evaluation")
 
+        # Setup Sharded DDP training
+        self.sharded_dpp = False
+        if args.sharded_ddp:
+            if args.local_rank == -1:
+                logger.warn("Using sharded DDP only works in distributed training.")
```

Comment:
I'd assert here - since clearly the user wanted to deploy this option - in general I find that using warnings to communicate obvious conflicts doesn't serve the purpose and wastes user's time.